modules/stochastic_tools/src/functions/DRLRewardFunction.C

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "DRLRewardFunction.h"
+#include "FunctionInterface.h"
+
+registerMooseObject("StochasticToolsApp", DRLRewardFunction);
+
+InputParameters
+DRLRewardFunction::validParams()
+{
+  InputParameters params = Function::validParams();
+
+  params.addClassDescription("Evaluates a reward function for a process which is controlled by a "
+                             "Deep Reinforcement Lerning based surrogate.");
+
+  params.addRequiredParam<FunctionName>("design_function", "The desired value to reach.");
+  params.addRequiredParam<PostprocessorName>(
+      "observed_value", "The name of the Postprocessor that contains the observed value.");
+  params.addParam<Real>("c1", 10, "1st coefficient in the reward function.");
+  params.addParam<Real>("c2", 1, "2nd coefficient in the reward function.");
+
+  return params;
+}
+
+DRLRewardFunction::DRLRewardFunction(const InputParameters & parameters)
+  : Function(parameters),
+    FunctionInterface(this),
+    _design_function(getFunction("design_function")),
+    _observed_pp_name(getParam<PostprocessorName>("observed_value")),
+    _observed_value(getPostprocessorValueByName(_observed_pp_name)),
+    _c1(getParam<Real>("c1")),
+    _c2(getParam<Real>("c2"))
+{
+}
+
+Real
+DRLRewardFunction::value(Real t, const Point & p) const
+{
+  Real design_value = _design_function.value(t, p);
+  return -_c1 * std::abs(design_value - _observed_value) + _c2;
+}
+
+ADReal
+DRLRewardFunction::value(const ADReal & t, const ADPoint & p) const
+{
+  ADReal design_value = _design_function.value(t, p);
+  return -_c1 * std::abs(design_value - _observed_value) + _c2;
+}
```

Comment:
The `-_c1 * std::abs(design_value - _observed_value) + _c2` here is just one simple example of the reward function in DRL. So I would not call it `DRLRewardFunction`. Maybe something more descriptive of the form, e.g., `ScaledAbsDifferenceDRLRewardFunction`.

It maybe better to have something like the `DRLRewardFunctionBase` or `RewardFunctionBase` where it only stores the `design_value` and the `_observed_value`. Then have the actual form (`value()`) of the reward function defined in a child class that is named based on the function formulation. This would allow definitions of different reward function formulations for other applications.

Reply:
I'll just go ahead and rename it, considering that the reward values come in the training process through a postprocessor which is agnostic of the function type. I don't think we will need a base class here.